src/sentry/relay/config/experimental.py

```diff
@@ -0,0 +1,88 @@
+import logging
+import time
+from collections.abc import MutableMapping
+from typing import Any, Protocol
+
+import sentry_sdk
+
+logger = logging.getLogger(__name__)
+
+
+class TimeoutException(BaseException):
+    _elapsed: float
+    _timeout: float
+
+    def __init__(self, elapsed: float, timeout: float, *args: object) -> None:
+        super().__init__(*args)
+        self._elapsed = elapsed
+        self._timeout = timeout
+
+
+class BuildTimeChecker:
```

Comment:
You could implement this as a coroutine (not saying you should):

```python
def time_checker(timeout: timedelta):
    start = now()
    while True:
        yield
        if (now() - start) > timeout:
            raise Timeout

def main():
    checker = time_checker(...)
    for _ in range(100):
        do_something()
        next(checker)

```

Reply:
Thanks for the suggestion! I don't really like the coroutine in this case: I don't find DX to be better, it's harder to extend `check` in the future and it requires an initial `next()` to start the coroutine. I updated the code to use `timedelta`s though.